src/shim/index.js

```diff
@@ -122,6 +122,18 @@ const commands = {
 function read() {
     let line = system.stdin.readLine();
     if (line) {
+        if (line === 'NOOP') {
```

Comment:
If `line === NOOP` then why do you have to respond with `NOOP`. This ain't no PING PONG! :) Can you just ignore it and let the events pass through. I am pretty sure that it was doing this before.